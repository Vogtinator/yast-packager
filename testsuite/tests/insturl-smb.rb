# encoding: utf-8

module Yast
  class InsturlSmbClient < Client
    def main
      Yast.include self, "testsuite.rb"
      @READ = {
        "target" => { "size" => 1, "tmpdir" => "/tmp" },
        "etc"    => {
          "install_inf" => {
            "InstMode"  => "smb",
            "Server"    => "192.168.1.1",
            "Serverdir" => "pub",
            "Share"     => "SHARE"
          }
        }
      }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "InstURL"
      Yast.import "Linuxrc"

      TEST(lambda { InstURL.installInf2Url("") }, [@READ], nil)
      TEST(lambda { InstURL.installInf2Url("test") }, [@READ], nil)
      TEST(lambda { InstURL.installInf2Url("/test") }, [@READ], nil)


      Ops.set(@READ, ["etc", "install_inf", "Serverdir"], "/pub")
      Linuxrc.ResetInstallInf
      TEST(lambda { InstURL.installInf2Url("") }, [@READ], nil)

      Ops.set(@READ, ["etc", "install_inf", "Username"], "suse")
      Ops.set(@READ, ["etc", "install_inf", "Password"], "yast2")
      Ops.set(@READ, ["etc", "install_inf", "Serverdir"], "pub")
      Linuxrc.ResetInstallInf
      TEST(lambda { InstURL.installInf2Url("") }, [@READ], nil)

      Ops.set(@READ, ["etc", "install_inf", "WorkDomain"], "group")
      Linuxrc.ResetInstallInf
      TEST(lambda { InstURL.installInf2Url("") }, [@READ], nil)

      nil
    end
  end
end

Yast::InsturlSmbClient.new.main
