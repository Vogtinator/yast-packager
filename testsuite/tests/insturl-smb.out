Read	.target.size "/etc/install.inf" 1
Dir	.etc.install_inf: ["InstMode", "Server", "Serverdir", "Share"]
Read	.etc.install_inf."InstMode" "smb"
Read	.etc.install_inf."Server" "192.168.1.1"
Read	.etc.install_inf."Serverdir" "pub"
Read	.etc.install_inf."Share" "SHARE"
Return	smb://192.168.1.1/SHARE/pub
Return	smb://192.168.1.1/SHARE/pub/test
Return	smb://192.168.1.1/SHARE/pub/test
Read	.target.size "/etc/install.inf" 1
Dir	.etc.install_inf: ["InstMode", "Server", "Serverdir", "Share"]
Read	.etc.install_inf."InstMode" "smb"
Read	.etc.install_inf."Server" "192.168.1.1"
Read	.etc.install_inf."Serverdir" "/pub"
Read	.etc.install_inf."Share" "SHARE"
Return	smb://192.168.1.1/SHARE//pub
Read	.target.size "/etc/install.inf" 1
Dir	.etc.install_inf: ["InstMode", "Password", "Server", "Serverdir", "Share", "Username"]
Read	.etc.install_inf."InstMode" "smb"
Read	.etc.install_inf."Password" "yast2"
Read	.etc.install_inf."Server" "192.168.1.1"
Read	.etc.install_inf."Serverdir" "pub"
Read	.etc.install_inf."Share" "SHARE"
Read	.etc.install_inf."Username" "suse"
Return	smb://suse:yast2@192.168.1.1/SHARE/pub
Read	.target.size "/etc/install.inf" 1
Dir	.etc.install_inf: ["InstMode", "Password", "Server", "Serverdir", "Share", "Username", "WorkDomain"]
Read	.etc.install_inf."InstMode" "smb"
Read	.etc.install_inf."Password" "yast2"
Read	.etc.install_inf."Server" "192.168.1.1"
Read	.etc.install_inf."Serverdir" "pub"
Read	.etc.install_inf."Share" "SHARE"
Read	.etc.install_inf."Username" "suse"
Read	.etc.install_inf."WorkDomain" "group"
Return	smb://suse:yast2@192.168.1.1/SHARE/pub;workgroup=group
